### PR TITLE
Add GITSIGN_LOG environment variable

### PR DESCRIPTION
**Pull request by wlynch**
#### Summary

Add GITSIGN_LOG environment variable for debug log path.

Makes debug log behavior opt-in, and lets users control where the logs
are written.

Signed-off-by: Billy Lynch <billy@chainguard.dev>

#### Ticket Link

n/a

#### Release Note

```release-note
Environment variable GITSIGN_LOG can be set to log status output to file for debugging.
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,13 +67,29 @@ Date:   Mon May 2 16:51:44 2022 -0400
...
+## Debugging
+
+If there is an error during signing, you may see an error like:
+
+```
+error: gpg failed to sign the data
+fatal: failed to write commit object
+```
+
+Because of [`Limitations`](#limitations) with Git signing tools, since `gitsign`
```
Comment (imjasonh) [inline README.md:84]: Is it possible that sensitive contents could be logged? Could we make this opt-in, and opt in during our tests?

Comment (wlynch) [reply README.md:84]: I don't think we do, but there's not enforcement on that. How does an env variable for the path sound?

Comment (nsmith5) [reply README.md:84]: I like opt-in. Don't write things anywhere by default and e.g `--log oh-dear-what-has-gone-wrong.log` when things go bad. Or `GITSIGN_LOG=how-did-we-git-here.log`

Comment (wlynch) [reply README.md]: Done!

**Review by nsmith5 — APPROVED** [review]

**Pull request merged**